app/src/test/java/com/example/marcinwisniewski/intivetrainingapp/RxImmediateSchedulerRule.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.marcinwisniewski.intivetrainingapp
+
+import io.reactivex.android.plugins.RxAndroidPlugins
+import io.reactivex.plugins.RxJavaPlugins
+import io.reactivex.schedulers.Schedulers
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+class RxImmediateSchedulerRule : TestRule {
```

Comment:
I think this is quite nice solution. But in my opinion it's better to create some kind of wrapper for schedulers and create additional wrapper for tests. But it's up to you. If you prefer test rule, that's fine.

Reply:
i will stay with this method but I will try to create wrapper also to learn how to do it